File: app/src/main/java/com/ozhan/mustafa/howl/core/users/add/AddUserContract.java
package com.ozhan.mustafa.howl.core.users.add;

import android.content.Context;

import com.google.firebase.auth.FirebaseUser;

/**
 * Created by Mustafa Özhan on 3/4/17 at 12:58 PM.
 */

public interface AddUserContract {
    interface View {
        void onAddUserSuccess(String message);

        void onAddUserFailure(String message);
    }

    interface Presenter {
        void addUser(Context context, FirebaseUser firebaseUser);
    }

    interface Interactor {
        void addUserToDatabase(Context context, FirebaseUser firebaseUser);
    }

    interface OnUserDatabaseListener {
        void onSuccess(String message);

        void onFailure(String message);
    }
}

File: app/src/main/java/com/ozhan/mustafa/howl/core/users/add/AddUserContract.kt
package com.ozhan.mustafa.howl.core.users.add

import android.content.Context

import com.google.firebase.auth.FirebaseUser

/**
 * Created by Mustafa Özhan on 3/4/17 at 12:58 PM.
 */

interface AddUserContract {
    interface View {
        fun onAddUserSuccess(message: String)

        fun onAddUserFailure(message: String)
    }

    interface Presenter {
        fun addUser(context: Context, firebaseUser: FirebaseUser)
    }

    interface Interactor {
        fun addUserToDatabase(context: Context, firebaseUser: FirebaseUser)
    }

    interface OnUserDatabaseListener {
        fun onSuccess(message: String)

        fun onFailure(message: String)
    }
}